adapter/test/posix/posix_adapter_basic_test.cpp

```diff
@@ -535,8 +535,8 @@ TEST_CASE("BatchedReadStrideDynamic",
     REQUIRE(fd != -1);
     std::string data(args.request_size, '1');
     for (size_t i = 0; i < info.num_iterations; ++i) {
-      size_t offset = abs(((i * rand_r(&info.offset_seed)) % info.stride_size) %
-                          info.total_size);
+      size_t offset = abs((int)(((i * rand_r(&info.offset_seed))
+                                 % info.stride_size) % info.total_size));
```

Comment:
You make the same change to the same code in 4 different places in this file. Why not make a helper function `size_t GetRandomOffset(size_t i, Info info);` so you have this code in one place?

Reply:
Done

Reply:
You define `GetRandomOffset` four times, which defeats the purpose. Can't you define it just once, pass the seed and stride size as parameters, and use the same function everywhere?

Reply:
I created a utility header file where I moved the function and called it from there.